src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -4986,9 +5101,22 @@ private String getChronologyName(Chronology chrono, Locale locale) {
          * @param timeStyle  the time style to use, may be null
          */
         LocalizedPrinterParser(FormatStyle dateStyle, FormatStyle timeStyle) {
```

Comment:
Can the constructors be `private`.  
The combination of package protected and the style of caller doing the validation makes me a bit nervous.
There should not be any callers outside of DateTimeFormatterBuilder.